ams/CleanupCommand.cs

```diff
@@ -0,0 +1,233 @@
+﻿using AMSMigrate.Ams;
+using AMSMigrate.Contracts;
+using AMSMigrate.Transform;
+using Azure;
+using Azure.Core;
+using Azure.ResourceManager.Media;
+using Azure.Storage.Blobs;
+using Microsoft.Extensions.Logging;
+using Microsoft.Identity.Client.Extensions.Msal;
+using Spectre.Console;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace AMSMigrate.ams
+{
+    internal class CleanupCommand : BaseMigrator
+    {
+        private readonly ILogger _logger;
+        private readonly CleanupOptions _options;
+        private readonly IMigrationTracker<BlobContainerClient, AssetMigrationResult> _tracker;
+
+        public CleanupCommand(GlobalOptions globalOptions,
+            CleanupOptions cleanupOptions,
+            IAnsiConsole console,
+            TokenCredential credential,
+            IMigrationTracker<BlobContainerClient, AssetMigrationResult> tracker,
+            ILogger<CleanupCommand> logger)
+            : base(globalOptions, console, credential)
+        {
+            _options = cleanupOptions;
+            _logger = logger;
+            _tracker = tracker;
+        }
+        public override Task MigrateAsync(CancellationToken cancellationToken)
+        {
+            return Task.FromResult<object>(null);
+        }
+
+        public async Task CleanUpAsync(CancellationToken cancellationToken)
+        {
+                var account = await GetMediaAccountAsync(_options.AccountName, cancellationToken);
+                Dictionary<string, bool> stats = new Dictionary<string, bool>();
+
+                    var totalAssets = await QueryMetricAsync(
+                        account.Id.ToString(),
+                    "AssetCount",
+                        cancellationToken: cancellationToken);
+
+                    _logger.LogInformation("The total asset count of the media account is {count}.", totalAssets);
+            AsyncPageable<MediaAssetResource>  assets = null;
+            if (!_options.IsCleanUpAccount)
+            {
+                var resourceFilter = GetAssetResourceFilter(_options.ResourceFilter, null, null);
+
+                var orderBy = "properties/created";
+                assets = account.GetMediaAssets()
+                    .GetAllAsync(resourceFilter, orderby: orderBy, cancellationToken: cancellationToken);
+            }
+            else
+            {
+                assets = account.GetMediaAssets()
+                   .GetAllAsync();
+            }
+            List<MediaAssetResource>? assetList = await assets.ToListAsync(cancellationToken);
+
+                    foreach (var asset in assetList)
```

Comment:
Nit:  Tab issue

Reply:
There are lots of Tab issues in this source file, you may ensure to config the VS or VS code appropriately so that the right number of spaces are inserted for the tabs.